documentation/general/workloads/cross-version-workloads.md

```diff
@@ -0,0 +1,138 @@
+    
+    end
+    
+    wasm-tools.7.0.100----->|extends|wasm-tools-net7-abstract
+    wasm-tools-net6----->|extends|wasm-tools-net6-abstract
+    
+    subgraph Mono.ToolChain.net7
+    wasm-tools-net7-abstract([wasm-tools-net7-abstract])-->Mono.browser-wasm.net7
+    Mono.browser-wasm.net7-->|alias|Mono.browser-wasm.v7["Mono.browser-wasm v7.0.0"]
+    wasm-tools-net7-abstract-->AOT.Cross.browser-wasm.net7
+    AOT.Cross.browser-wasm.net7-->|alias|AOT.Cross.browser-wasm.v7["AOT.Cross.browser-wasm v7.0.0"]
+    end
+    
+    wasm-tools-net7-abstract------>|extends|microsoft-net-sdk7-emscripten;
+    
+    subgraph workload.emscripten.net7
+    microsoft-net-sdk7-emscripten(["microsoft-net7-sdk-emscripten<BR><I>(abstract)</I>"])-->Microsoft.NET.Runtime.7.Emscripten.Node[Emscripten.Node.net7];
+    Microsoft.NET.Runtime.7.Emscripten.Node-->|alias|Microsoft.NET.Runtime.Emscripten.Node["Emscripten.Node v7.0.0"]
+    microsoft-net-sdk7-emscripten-->Microsoft.NET.Runtime.7.Emscripten.Python[Emscripten.Python.net7];
+    Microsoft.NET.Runtime.7.Emscripten.Python-->|alias|Microsoft.NET.Runtime.Emscripten.Python["Emscripten.Python v7.0.0"]
+    microsoft-net-sdk7-emscripten-->Microsoft.NET.Runtime.7.Emscripten.Sdk[Emscripten.Sdk.net7];
+    Microsoft.NET.Runtime.7.Emscripten.Sdk-->|alias|Microsoft.NET.Runtime.Emscripten.Sdk["Emscripten.Sdk v7.0.0"]
+    end
+
+```
+
+## Aliased workload packs
+
+Workload pack aliases will be used to support depending on different versions of the same workload pack ID.  This is similar to the [Side-by-Side Workload Pattern](https://github.com/dotnet/designs/blob/main/accepted/2020/workloads/workload-manifest.md#side-by-side-workload-pattern) described in the workload manifest design spec.  Using `Emscripten.Sdk` as an example, there will be a version 6.0.0 and a version 7.0.0 of the workload pack.  The `workload.emscripten.net6` manifest will alias `Emscripten.Sdk.net6` to version 6.0.0 of the package, and the `workload.emscripten.net7` manifest will alias `Emscripten.Sdk.net7` to version 7.0.0 of the package.  Then the logic which imports the workload pack can conditionally import the right version of the pack based on the target framework version.  This logic is expected to be in the `WorkloadManifest.targets` file for the workload manifest that defines the `wasm-tools` workload.  For the 7.0.100 version of that manifest, the import logic would look something like this:
+
+```xml
+  <ImportGroup Condition="'$(RunAOTCompilation)' == 'true' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(TargetFrameworkVersion)' == '6.0'">
+    <Import Project="Sdk.props" Sdk="Emscripten.Sdk.net6" />
+  </ImportGroup>
+  <ImportGroup Condition="'$(RunAOTCompilation)' == 'true' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(TargetFrameworkVersion)' == '7.0'">
+    <Import Project="Sdk.props" Sdk="Emscripten.Sdk.net7" />
+  </ImportGroup>
+```
```

Comment:
When .NET 7 is stable, do we anticipate that building a `net6.0-android` project would use the .NET 7 Android workload (SDK & .targets)? Or will it somehow need to use the *old* .NET 6 targets?

Reply:
It should use the .NET 7 MSBuild tasks and targets from the workload, and they should be written so that they support targeting .NET 6 and .NET 7.

(A workload could decide to use the .NET 6 targets, but I wouldn't recommend that because it could be hard to keep multiple SDK versions and multiple workload targets all compatible with each other).